### PR TITLE
Various refactors and tidy up.

### PR DESCRIPTION
**Pull request by wg4568**
The biggest change here is to the file structure.

- game/ is for source code
- game/lib is for internal libraries and APIs (this is a Python module, can be imported from game/)
- res/ is for resources, static files etc. Will need to be subdivided into types as the project grows.
- towergame.bat is just a shortcut for running game/main.py.

The second biggest change is using a code formatter. I went with [Black](https://black.readthedocs.io/en/stable/). You can install it using `pip install black` and configure vscode to use it by running "format document" and opting to use Black (a popup in the bottom right). I would also recommend enabling "format on save" in settings.

### CODE REVIEW

**Comment by wg4568** [comment]
Just a note, I made no changes to the code itself.

**Comment by wg4568** [comment]
Just did some reading, we might want to configure our pylint.yml file to use Black also. It's a little more opinionated (and therefor more consistent) when compared to the default pylint stuff. It looks like @AdamJCavanaugh has done some work on this

**Comment by AdamJCavanaugh** [comment]
Yeah, I normally use flake8, but black would be fine. I'll plan to use that when I get around to working on the action.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
-__pycache__
-.vscode
-tools
\ No newline at end of file
+.vscode/
+.venv/
+__pycache__/
+tools/
```
Comment (AdamJCavanaugh) [inline .gitignore:4]: Hey, what do you have in tools? Anything nice? :)

**Review by AdamJCavanaugh — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (jamesg31) [reply .gitignore:4]: > Hey, what do you have in tools? Anything nice? :)

Oh no you caught me! That is just where I have a couple janky scripts to scrape the data for the airport.hjson file. You didnt think I made that by hand xD

Comment (AdamJCavanaugh) [reply .gitignore:4]: I _hoped_ you didn't! lol

**Pull request merged**